x/fswap/types/genesis.go

```diff
@@ -1,24 +1,28 @@
 package types
 
-import (
-// this line is used by starport scaffolding # genesis/types/import
-)
+import sdk "github.com/Finschia/finschia-sdk/types"
 
-// DefaultIndex is the default capability global index
-const DefaultIndex uint64 = 1
+// NewGenesis creates a new genesis state
+func NewGenesisState(config Config, swappable sdk.Coin) *GenesisState {
+	return &GenesisState{
+		Params:  NewParams(swappable),
+		Swapped: NewSwapped(config),
+	}
+}
 
 // DefaultGenesis returns the default Capability genesis state
 func DefaultGenesis() *GenesisState {
-	return &GenesisState{
-		// this line is used by starport scaffolding # genesis/types/default
-		Params: DefaultParams(),
-	}
+	return NewGenesisState(DefaultConfig(), DefaultSwappableNewCoinAmount)
 }
 
 // Validate performs basic genesis state validation returning an error upon any
 // failure.
 func (gs GenesisState) Validate() error {
-	// this line is used by starport scaffolding # genesis/types/validate
-
-	return gs.Params.Validate()
+	if err := gs.Params.Validate(); err != nil {
+		return err
+	}
+	if err := gs.Swapped.Validate(); err != nil {
+		return err
+	}
```

Comment:
The denoms are injected through `NewKeeper()` which allows the application developer to change the denominations. However, this validation logic lacks related verification, which leads to mismatch between the default genesis state and the configuration.
The problem is originated from `Coin` in the state. I recommend you address this issue.